gateway/gateway.go

```diff
@@ -788,11 +788,17 @@ func (gateway *HandleT) isUserSuppressed(workspaceID, userID, sourceID string) b
 	if !enableSuppressUserFeature || gateway.suppressUserHandler == nil {
 		return false
 	}
-	return gateway.suppressUserHandler.IsSuppressedUser(
+	metadata := gateway.suppressUserHandler.GetSuppressedUser(
 		workspaceID,
 		userID,
 		sourceID,
 	)
+	if metadata != nil {
+		createdAt := metadata.CreatedAt
+		userSuppressionTimer := gateway.stats.NewTaggedStat("gateway.user_suppression_age", stats.TimerType, stats.Tags{"workspaceId": workspaceID, "sourceID": sourceID})
+		userSuppressionTimer.Since(createdAt)
+	}
+	return metadata != nil
```

Comment:
```suggestion
	if metadata := gateway.suppressUserHandler.GetSuppressedUser(workspaceID, userID, sourceID); metadata != nil {
		gateway.stats.NewTaggedStat("gateway.user_suppression_age", stats.TimerType, stats.Tags{
			"workspaceId": workspaceID,
			"sourceID":    sourceID,
		}).Since(metadata.CreatedAt)
		return true
	}
	return false
```

Reply:
Done